environment-dev.yml

```diff
@@ -30,43 +36,47 @@ dependencies:
   - pandas-gbq
   - pytables
   - msgpack-python
-  - psutil
-  - pytest>=6.0.1
-  - pytest-benchmark
-  - pytest-cov>=2.10.1
-  - pytest-xdist>=2.1.0
-  - packaging
-  - coverage
-  - pygithub
-  - rpyc==4.1.5
-  - cloudpickle
-  - boto3
-  - moto
   - scikit-learn
   - pymssql
   - psycopg2
-  # Mypy 0.990 doesn't work: https://github.com/modin-project/modin/issues/5206  
-  - mypy!=0.990
-  - pandas-stubs
   - fastparquet<2023.1.0
-  # for release script
-  - pygit2
   # TODO(https://github.com/modin-project/modin/issues/5194): Uncap xgboost
   # when we use collective instead of rabit.
   - xgboost>=1.7.1,<2.0.0
   - tqdm
+
+  ## modin in the cloud dependencies
+  - boto3
+  - cloudpickle
+  - rpyc==4.1.5
+
+  # dependencies for making release
+  - pygithub>=v1.58.0
+  - pygit2>=1.9.2
+
+  # test dependencies
+  - coverage>=7.1.0
+  - moto>=4.1.0
+  - pytest>=7.2.1
+  - pytest-benchmark>=4.0.0
+  - pytest-cov>=4.0.0
+  - pytest-xdist>=3.2.0
+
   # code linters
-  - black
-  - flake8
-  - flake8-no-implicit-concat
-  - flake8-print
+  - black>=23.1.0
+  - flake8>=6.0.0
+  - flake8-no-implicit-concat>=0.3.4
+  - flake8-print>=5.0.0
+  - mypy>=1.0.0
+  - pandas-stubs>=2.0.0
+
   - pip:
+      # no conda package for windows so we install it with pip
+      - connectorx>=0.2.6a4
+      # experimental version of fuzzydata requires at least 0.0.6 to successfully resolve all dependencies
+      - fuzzydata>=0.0.6
       # Fixes breaking ipywidgets changes, but didn't release yet.
       - git+https://github.com/modin-project/modin-spreadsheet.git@49ffd89f683f54c311867d602c55443fb11bf2a5
       - git+https://github.com/airspeed-velocity/asv.git@ef016e233cb9a0b19d517135104f49e0a3c380e9
```

Comment:
why do we need `asv` here? aren't we installing it via `pip` in the CI anyway?

Reply:
To make it easier for the user to install this library, however, not everyone needs it. Maybe the installation of this library should be transferred to another optional extras in `setup.py`? (for example, `perf`)

Reply:
Let's think on splitting the requirements for developing (run by people) and for running tests (run by automation), most likely in a separate issue.
I have some ideas on how to move forward, but I need some time to properly formulate those...

Reply:
ok!